### PR TITLE
ls: silently ignore `-T` option

### PR DESCRIPTION
**Pull request by Stonks3141**
The option can't be implemented without a significant rewrite according to #3624, and right now passing the option causes a failure.

* fixed error when `-T` option is passed
* write tests for `-T` option

### CODE REVIEW

**Comment by sylvestre** [comment]
i don't think we can mix clap 3.1 & 3.2
see
https://github.com/uutils/coreutils/issues/3634#issuecomment-1154870604=

**Comment by Stonks3141** [comment]
Ok, I reverted the changes and removed the relevant tests.

**Comment by sylvestre** [comment]
many CI jobs are failing, could you please have a look?
thanks

**Comment by Stonks3141** [comment]
They all say
```text
error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.64.0-nightly (23e21bdd2 2022-07-15) running on x86_64-unknown-linux-gnu

note: compiler flags: --crate-type lib -C embed-bitcode=no -C debuginfo=2 -C incremental -Z binary-dep-depinfo -Z save-analysis
```
I don't think I changed anything that would break CI, I just added a Clap arg and a test function. I can replicate the error locally, but the error is in `uu_od`. It also occurs if I clone the upstream repo and run `cargo +nightly udeps` (udeps needs nightly).

**Comment by Stonks3141** [comment]
Most of the CI jobs magically fixed themselves, and I fixed the style/spelling issues. Should be ready to merge now, unless you have any other issues?

File: tests/by-util/test_ls.rs [hunk 1]
```diff
@@ -3117,3 +3117,13 @@ fn test_dereference_symlink_file_color() {
         .succeeds()
         .stdout_is(out_exp);
 }
+
+#[test]
+fn test_tab_size_option() {
+    let scene = TestScenario::new(util_name!());
+
+    scene.ucmd().args(&["-T", "3"]).succeeds();
+
```
Comment (sylvestre) [inline tests/by-util/test_ls.rs:3126]: please run rustfmt on this file too

Comment (Stonks3141) [reply tests/by-util/test_ls.rs]: Should be done, there was a trailing newline.

File: tests/by-util/test_ls.rs [hunk 2]
```diff
@@ -3117,3 +3117,12 @@ fn test_dereference_symlink_file_color() {
         .succeeds()
         .stdout_is(out_exp);
 }
+
+#[test]
+fn test_tabsize_option() {
+    let scene = TestScenario::new(util_name!());
+
+    scene.ucmd().args(&["-T", "3"]).succeeds();
```
Comment (sylvestre) [inline tests/by-util/test_ls.rs:3125]: I would also like to see a new test with what it should actually return once it is implemented
and marked as ignored.
Could you please do that ? thanks

Comment (Stonks3141) [reply tests/by-util/test_ls.rs:3125]: Ok, done.

Comment (Stonks3141) [reply tests/by-util/test_ls.rs:3125]: Spellcheck doesn’t like “aaaaaaaa”, “bbbb”, etc. Should I just add them to the spell ignore?

Comment (sylvestre) [reply tests/by-util/test_ls.rs:3125]: yeah

**Pull request merged**